src/server/rdb_load.cc

```diff
@@ -465,9 +467,14 @@ void RdbLoaderBase::OpaqueObjLoader::operator()(const JsonType& json) {
 }
 
 void RdbLoaderBase::OpaqueObjLoader::CreateSet(const LoadTrace* ltrace) {
-  size_t len = ltrace->blob_count();
+  size_t increment = 1;
+  if (rdb_type_ == RDB_TYPE_SET_WITH_EXPIRY) {
+    increment = 2;
+  }
+
+  size_t len = ltrace->blob_count() / increment;
 
-  bool is_intset = true;
+  bool is_intset = (rdb_type_ == RDB_TYPE_HASH);
   if (len <= SetFamily::MaxIntsetEntries()) {
```

Comment:
I think the condition (rdb_type_ == RDB_TYPE_HASH && ltrace->blob_count() < SetFamily::MaxIntsetEntries()) will be more clear here.
you will not need the increment logic above

Reply:
I do need the increment logic for incrementing the loop, but I moved that below. Otherwise, SG.